j5/components/base.py

```diff
@@ -16,3 +16,9 @@ class Component(metaclass=ABCMeta):
     def interface_class() -> Type[Interface]:
         """Get the interface class that is required to use this component."""
         raise NotImplementedError  # pragma: no cover
+
+
+class NotSupportedByHardware(Exception):
```

Comment:
I think this is is an error and as such such be [suffixed with Error](https://www.python.org/dev/peps/pep-0008/#exception-names), however I don't have a super strong opinion on it if you or others disagree.

Reply:
Also, a question that's outside of the scope of this PR, should we have a base J5Exception that we use as our base Exception class, so that consumers can optionally just catch any of our stuff throwing?

Reply:
I think that's a good idea, and have opened #142 to track it